src/Splits.sol

```diff
@@ -25,32 +25,23 @@ abstract contract Splits {
     uint32 internal constant _TOTAL_SPLITS_WEIGHT = 1_000_000;
     /// @notice The amount the contract can keep track of each ERC-20 token.
     // slither-disable-next-line unused-state
-    uint128 internal constant _MAX_SPLITS_BALANCE = type(uint128).max;
+    uint128 internal constant _MAX_SPLITS_BALANCE = _SPLITTABLE_MASK;
```

Comment:
It is not only a PR for gas optimization it limits the max split balance to `type(uint128).max/2`.

Reply:
Yes, that's a good observation. It's not visible on the outside of the protocol though. We have a test in place checking that `_MAX_SPLITS_BALANCE` isn't smaller than the protocol-wide `MAX_TOTAL_BALANCE`.